src/policies/goals.js

```diff
@@ -1,38 +1,67 @@
-/*
-  This class handles authorization for Activity Reports. In express handlers this
-  object should be constructed with the current user and the activity report to be
-  updated/retrieved for update/get or the new report for create. The permissions
-  of the user are checked against the report and simple questions about the actions
-  the user can take on that report are answered, mainly can the user create, update
-  or get the report.
-*/
-import _ from 'lodash';
+import { find, isUndefined } from 'lodash';
 import SCOPES from '../middleware/scopeConstants';
 
 export default class Goal {
-  constructor(user, goal) {
+  constructor(user, goal, regionId) {
     this.user = user;
     this.goal = goal;
+    this.regionId = regionId;
   }
 
+  // this expects goal to have been found with associated data, specifically
+  // goalByIdWithActivityReportsAndRegions in services/goals
+  // you can see the structure expected in the conditions below
   canDelete() {
-    return this.canWriteInRegion();
+    if (this.isOnApprovedActivityReports()) {
+      return false;
+    }
+
+    // I assume that if you can delete in ANY of the goals regions,
+    // than you can delete it
+    const regions = this.goal.grants.map((grant) => grant.regionId);
+    return regions.reduce((previous, region) => {
```

Comment:
Can we use `regions.every` here instead of the reducer? If so it might be a bit easier to understand what is going on, and probably can remove the explicit true check on line 24. Not a big deal either way, especially since this code is going to be updated very soon after the schema update.

Reply:
I think we want ```regions.some```, right? What I'm asking is, if a user can delete in any of the regions then should they be able to delete here? Or would they need permissions in all a goals regions? (I know its almost a non-issue given how few goals transcend 1 region, but just checking in case)

Reply:
Yeah you are correct it doesn't really matter because this code isn't making it to release with the schema updates. But I'd typically land on the side of not letting users do things when there are permission questions like this that could go either way. Rather be more constrained then more permissive when letting users mutate data.